src/core/DataFile.cpp

```diff
@@ -89,18 +89,27 @@ const std::vector<ProjectVersion> DataFile::UPGRADE_VERSIONS = {
 	"1.2.0-rc3"        ,   "1.3.0"
 };
 
-std::array<DataFile::typeDescStruct, DataFile::TypeCount> DataFile::s_types =
+namespace 
 {
-	DataFile::typeDescStruct{ DataFile::UnknownType, "unknown" },
-	DataFile::typeDescStruct{ DataFile::SongProject, "song" },
-	DataFile::typeDescStruct{ DataFile::SongProjectTemplate, "songtemplate" },
-	DataFile::typeDescStruct{ DataFile::InstrumentTrackSettings, "instrumenttracksettings" },
-	DataFile::typeDescStruct{ DataFile::DragNDropData, "dnddata" },
-	DataFile::typeDescStruct{ DataFile::ClipboardData, "clipboard-data" },
-	DataFile::typeDescStruct{ DataFile::JournalData, "journaldata" },
-	DataFile::typeDescStruct{ DataFile::EffectSettings, "effectsettings" },
-	DataFile::typeDescStruct{ DataFile::MidiClip, "midiclip" }
-} ;
+	struct LMMS_EXPORT typeDescStruct
```

Comment:
I realise this is how it was before, so feel free to consider this out of scope, but `LMMS_EXPORT` can be removed since this struct is only used in this file. (Also, our coding conventions require types to start with an uppercase letter, and including "struct" in the name of a struct feels weird to me.)